internal/input/assets.go

```diff
@@ -0,0 +1,64 @@
+package input
+
+import (
+	ingestio "github.com/stellar/go/ingest/io"
+	"github.com/stellar/go/xdr"
+	"github.com/stellar/stellar-etl/internal/utils"
+)
+
+// GetPaymentOperations returns a slice of payment operations that can include new assets from the ledgers in the provided range (inclusive on both ends)
+func GetPaymentOperations(start, end uint32, limit int64) ([]OperationTransformInput, error) {
+	backend, err := utils.CreateBackend()
+	if err != nil {
+		return []OperationTransformInput{}, err
+	}
+
+	defer backend.Close()
+
+	latestNum, err := backend.GetLatestLedgerSequence()
+	if err != nil {
+		return []OperationTransformInput{}, err
+	}
+
+	err = validateLedgerRange(start, end, latestNum)
+	if err != nil {
+		return []OperationTransformInput{}, err
+	}
+
+	opSlice := []OperationTransformInput{}
+	for seq := start; seq <= end; seq++ {
+		txReader, err := ingestio.NewLedgerTransactionReader(backend, publicPassword, seq)
+		if err != nil {
+			return []OperationTransformInput{}, err
+		}
+
+		for int64(len(opSlice)) < limit || limit < 0 {
```

Comment:
Why are we handling `limit < 0` or `limit >=0` in this loop? This seems like it's passed in and not modified in this loop. So, we can just early return if `limit < 0` before we even enter this loop. This doesn't change any function, but it does result in a cleaner approach with fewer conditions.

Reply:
This loop is reading each transaction at the ledger. Limit isn't changed, but if the length of the slice reaches limit, then we want to stop reading. Similarly, within a transaction there can be multiple operations, so we need that second check while iterating over the operations in the tx.Envelope.

Reply:
Ah, I was a bit unclear. I agree that we need the second check, for the reasons you stated. I'm not sure why we need to check `limit < 0`.

Reply:
If we have a negative limit, then that means we export everything. This statement is saying while `we are under the limit`, or `the limit is negative/we want to export everything`, keep exporting.